### PR TITLE
Refactored i/o checks and power states checks

### PR DESCRIPTION
**Pull request by m-hilgendorf**
- New module, ocdb/utils/netlist-checks/all with check-netlist(...) entrypoint
- Fixed a handful of bugs/edge cases in digital i/o checks
- Rewrote error messages for concision and precision
- Removed global state from netlist checking

### CODE REVIEW

File: utils/netlist-checks/power-states.stanza [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+defpackage ocdb/utils/netlist-checks/power-states :
+  import core
+  import collections
+  import jitx
+  import jitx/commands
+  import lang-utils
+  import ocdb/utils/netlist-checks/utils
+  
+
+public defn check-power-states (module:JITXObject, connections:Connections) :
```
Comment (m-hilgendorf) [inline utils/netlist-checks/power-states.stanza:11]: TODO: this needs to be updated as well, but it's not a pressing priority at the moment.

File: utils/netlist-checks/io-checks.stanza [hunk 2]
```diff
@@ -0,0 +1,508 @@
...
+    "Pins %, are marked OpenCollector on the same net and have %_" 
+      % [seq(ref{pin(_)}, open-collectors), action]
+
+defn open-collector-fail-message (open-collectors:Tuple<DigitalPin>, action:?) :
+  if length(open-collectors) == 1 :
+    "Pin %_ is marked OpenCollector but does not have %_" 
+      % [ref $ pin(open-collectors[0]), action]
+  else :
+    "Pins %, are marked OpenCollector on the same net but do not have %_" 
+      % [seq(ref{pin(_)}, open-collectors), action]
```
Comment (ariel-hi) [inline utils/netlist-checks/io-checks.stanza:92]: I like the emphasis on messages having proper grammar

File: utils/netlist-checks/io-checks.stanza [hunk 3]
```diff
@@ -0,0 +1,508 @@
...
+; Compute the parallel resistance of a tuple of resistors. 
+; Note on numerical stability : 
+; 
+; In the empty array case, we return false (instead of +INFINITY). 
+; Negative resistances are forbidden and will cause a fatal error.
+; 
+; Zero-ohm resistors in parallel will result in 0 total resistance.
+;
+defn parallel-resistance (resistors:Tuple<Instance>) -> Double :
+  val resistances = map({property(_.resistance) as Double}, resistors)
```
Comment (ariel-hi) [inline utils/netlist-checks/io-checks.stanza:258]: `resistance` property guaranteed by prior call to `pullup-resistors`, nice!

**Review by ariel-hi — APPROVED** [review]
Nice! I like the file separation too. Will require minor updates when Toleranced is moved.

File: utils/netlist-checks/io-checks.stanza (hunk 3 above)
Comment (m-hilgendorf) [reply utils/netlist-checks/io-checks.stanza:258]: Which is why this helper should stay private...

**Pull request merged**